### ISSUE TITLE
`explicit-length-check`: false positive when checking presence of property in TypeScript

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#2217]
In TypeScript, for a property that could be null / undefined / number, we need to check if the property exists first before performing a comparison on it, like this:

```ts
const obj: { size: number | null } = { size: 123 /* hardcoded for example */ };
if (obj.size && obj.size > 0) {
  console.log();
}
```

But this gets flagged by [explicit-length-check](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/explicit-length-check.md) and autofixed to:

```ts
const obj: { size: number | null } = { size: 123 /* hardcoded for example */ };
if (obj.size > 0 && obj.size > 0) {
  console.log();
}
```

Which hits the original TypeScript error:

```pt
'obj.size' is possibly 'null'.ts(18047)
```

Some fix ideas:

1. If we had type-awareness, we could ignore this, but I'm assuming we don't.
2. We could ignore the pattern `obj.size && obj.size > 0` on the assumption that the first part of the expression is a presence check and that it would it would be redundant to autofix the first part of the expression to another comparison.

**Comment by fregante** [comment]
You can use `obj?.size > 0`

**Comment by fregante** [comment]
I can confirm this issue. [TS playground link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBCBGArAXDA3jCBLAXgU1TAFcBbePAJxgB8ZiAbemAXxgF51NcCYBGAJgDMMAPQAqGAAsAhhQAmoOXjkwAZiCp4AHtJIAHenhhiRLANwAoLKpgAKBIgB02fDAB8MAAwBKdBZgwoJAgho70IADmtt6WzBZWNvZIzty+aP6B4BAheGGR0bFAA)

Solution:

- disable this rule, or 
- ensure that size _is_ set as early as possible

Your type has essentially two types for the same value: `null | 0`

**Comment by fregante** [comment]
This would be sufficient for TypeScript:

```ts
const object: {size: number | undefined} = {size: 123 /* hardcoded for example */};

if (object.size) {
	console.log();
}
```

The bugfix:

- the rule should ignore nullable `size`/`length` properties
